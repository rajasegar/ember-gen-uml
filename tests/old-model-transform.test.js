/* globals test expect */
const transform = require("../src/transform");

test('should generate model uml properly', function() {
  const input = `
export default DS.Model.extend({
  title: DS.attr(),
  name: DS.attr(),
  birthday: DS.attr(),
})`;

  const uml = transform(input, 'Person');
  expect(uml).toMatchSnapshot();
});

test('should generate model uml with data types properly', function() {
  const input = `
export default DS.Model.extend({
  name: DS.attr('string'),
  age: DS.attr('number'),
  admin: DS.attr('boolean'),
  birthday: DS.attr('date'),
})`;

  const uml = transform(input, 'User');
  expect(uml).toMatchSnapshot();
});

test('should generate model uml with data types with default values properly', function() {
  const input = `
export default DS.Model.extend({
  username: DS.attr('string'),
  email: DS.attr('string'),
  verified: DS.attr('boolean', { defaultValue: false }),
  createdAt: DS.attr('date', {
    defaultValue() { return new Date(); }
  }),
})`;

  const uml = transform(input,'User');
  expect(uml).toMatchSnapshot();
});

test('should generate model uml with belongsTo relationship', () => {
  const input = `
export default DS.Model.extend({
  profile: DS.belongsTo('profile'),
})`;
  const uml = transform(input, 'User');
  expect(uml).toMatchSnapshot();
});

test('should generate model uml with hasMany relationship', () => {
  const input = `
export default DS.Model.extend({
  comments: DS.hasMany('comment'),
})`;
  const uml = transform(input, 'BlogPost');
  expect(uml).toMatchSnapshot();
});

test('should generate model uml with one-to-many reflexive reelations', () => {
  const input = `
export default DS.Model.extend({
  children: DS.hasMany('folder', { inverse: 'parent' }),
  parent: DS.belongsTo('folder', { inverse: 'children' }),
})`;

  expect(transform(input, 'Folder')).toMatchSnapshot();
});

test('should generate model uml with one-to-one reflexive relations', () => {
  const input = `
export default DS.Model.extend({
  name: DS.attr('string'),
  bestFriend: DS.belongsTo('user', { inverse: 'bestFriend' }),
})`;

  expect(transform(input, 'User')).toMatchSnapshot();
});

test('should generate model uml with reflexive relations without inverse', () => {
  const input = `
export default DS.Model.extend({
  parent: DS.attr('belongsTo', { inverse: null }),
})`;

  expect(transform(input,'Child')).toMatchSnapshot();
});
