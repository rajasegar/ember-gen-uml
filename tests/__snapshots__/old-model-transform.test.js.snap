// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate model uml properly 1`] = `
"
@startuml


class Person {
  +title
  +name
  +birthday
}

@enduml"
`;

exports[`should generate model uml properly for Model.extend 1`] = `
"
@startuml


class Model {
  +title
  +name
  +birthday
}

@enduml"
`;

exports[`should generate model uml with belongsTo relationship 1`] = `
"
@startuml
profile o-- \\"belongsTo\\" User 

class User {
  +profile
}

@enduml"
`;

exports[`should generate model uml with data types properly 1`] = `
"
@startuml


class User {
  +name: string
  +age: number
  +admin: boolean
  +birthday: date
}

@enduml"
`;

exports[`should generate model uml with data types with default values properly 1`] = `
"
@startuml


class User {
  +username: string
  +email: string
  +verified: boolean
  +createdAt: date
}

@enduml"
`;

exports[`should generate model uml with hasMany relationship 1`] = `
"
@startuml
comment *-- \\"hasMany\\" BlogPost 

class BlogPost {
  +comments
}

@enduml"
`;

exports[`should generate model uml with one-to-many reflexive reelations 1`] = `
"
@startuml
folder *-- \\"hasMany\\" Folder 
folder o-- \\"belongsTo\\" Folder 

class Folder {
  +children
  +parent
}

@enduml"
`;

exports[`should generate model uml with one-to-one reflexive relations 1`] = `
"
@startuml
user o-- \\"belongsTo\\" User 

class User {
  +name: string
  +bestFriend
}

@enduml"
`;

exports[`should generate model uml with reflexive relations without inverse 1`] = `
"
@startuml


class Child {
  +parent: belongsTo
}

@enduml"
`;
